samples/java-spring-cinema-booking-choreography/src/main/java/com/example/cinema/model/Show.java

```diff
@@ -0,0 +1,220 @@
+        int seatNumber = reserveSeat.seatNumber();
+        if (isDuplicate(reserveSeat.reservationId())) {
+            return left(DUPLICATED_COMMAND);
+        } else {
+            return seats.get(seatNumber).<Either<CinemaApiModel.ShowCommandError, ShowEvent>>map(seat -> {
+                if (seat.isAvailable()) {
+                    return right(new ShowEvent.SeatReserved(id, reserveSeat.walletId(), reserveSeat.reservationId(), seatNumber, seat.price(), availableSeats()-1));
+                } else {
+                    return left(SEAT_NOT_AVAILABLE);
+                }
+            }).getOrElse(left(SEAT_NOT_FOUND));
+        }
+    }
+
+    public Either<CinemaApiModel.ShowCommandError, ShowEvent> handleCancellation(CinemaApiModel.ShowCommand.CancelSeatReservation cancelSeatReservation) {
+        String reservationId = cancelSeatReservation.reservationId();
+        return pendingReservations.get(reservationId).fold(
+                () ->left(RESERVATION_NOT_FOUND),
+                /*matching reservation*/
+                seatNumber -> seats.get(seatNumber).<Either<CinemaApiModel.ShowCommandError, ShowEvent>>map(seat ->
+                        right(new ShowEvent.SeatReservationCancelled(id, reservationId, seatNumber, availableSeats()+1))
+                ).getOrElse(left(SEAT_NOT_FOUND))
+        );
+    }
+
+    public Either<CinemaApiModel.ShowCommandError, ShowEvent> handleConfirmation(CinemaApiModel.ShowCommand.ConfirmReservationPayment confirmReservationPayment) {
+        String reservationId = confirmReservationPayment.reservationId();
+        return pendingReservations.get(reservationId).fold(
+                () -> left(RESERVATION_NOT_FOUND),
+                seatNumber ->
+                        seats.get(seatNumber).<Either<CinemaApiModel.ShowCommandError, ShowEvent>>map(seat ->
+                                right(new ShowEvent.SeatReservationPaid(id, reservationId, seatNumber))
+                        ).getOrElse(left(SEAT_NOT_FOUND)));
+    }
+
+
+    private boolean isDuplicate(String reservationId) {
+        return pendingReservations.containsKey(reservationId) ||
+                finishedReservations.get(reservationId).isDefined();
+    }
+
+    public Show apply(ShowEvent event) {
+        return switch (event) {
+            case ShowEvent.ShowCreated ignored ->
+                    throw new IllegalStateException("Show is already created, use Show.create instead.");
+            case ShowEvent.SeatReserved seatReserved -> applyReserved(seatReserved);
+            case ShowEvent.SeatReservationPaid seatReservationPaid ->
+                    applyReservationPaid(seatReservationPaid);
+            case ShowEvent.SeatReservationCancelled seatReservationCancelled ->
+                    applyReservationCancelled(seatReservationCancelled);
+//            case ShowEvent.CancelledReservationConfirmed __ -> this;
+        };
+    }
+
+    public Show applyReserved(ShowEvent.SeatReserved seatReserved) {
```

Comment:
Here we could simply call this methods: `onEvent`: `Show onEvent(ShowEvent.SeatReserved seatReserved)`

applyReserved + SeatReserved is just repeating stuff and not adding much information. 

I would do the same for the command handlers.

`onCommand` and `onEvent` all the time. Otherwise we are just repeating names without adding much value.

Reply:
all these methods could be also private

Reply:
I agree but it was done like that for better visability for the reader